git-tool-providers/azure-dev-ops.mjs

```diff
@@ -1,33 +1,144 @@
-import * as azdev from "azure-devops-node-api";
-import * as dotenv from 'dotenv' 
-dotenv.config('../env');
+import * as azdev from 'azure-devops-node-api'
+import * as dotenv from 'dotenv'
+dotenv.config('../env')
+import { DateTime } from 'luxon'
+import { userMapping } from '../userMapping.mjs'
 
 const orgUrl = process.env.ORG_URL
 const token = process.env.AZURE_PERSONAL_ACCESS_TOKEN
-const repoId = process.env.REPO_ID
-const project = process.env.PROJECT_NAME
+const projects = process.env.PROJECT_NAMES.split(', ')
+const t = 'days between commits'
+const recent = `sum of all ${t}`
+const avgDays = `avg ${t}`
 
 export async function getAzureDevOpsProvider(orgUrl, token) {
-    let authHandler = azdev.getPersonalAccessTokenHandler(token); 
-    let webApi = new azdev.WebApi(orgUrl, authHandler);  
-    let gitApiObject = await webApi.getGitApi();
-    return gitApiObject;
+  let authHandler = azdev.getPersonalAccessTokenHandler(token)
+  let webApi = new azdev.WebApi(orgUrl, authHandler)
+  let gitApiObject = await webApi.getGitApi()
+
+  return gitApiObject
 }
 
 export async function getCommitersByName() {
-    let gitApiObject = await getAzureDevOpsProvider(orgUrl, token)
-    const commits = await gitApiObject.getCommits(repoId, {$skip: 0, $top: 15000}, project)
-    
-    if(commits && commits.length) {
-        let commiters = {}
-
-        commits.forEach((commit) => {
-            commiters[commit.author.name] ? 
-                commiters[commit.author.name].commitCount++ : commiters[commit.author.name] = { commitCount : 1 }
-        })
+  let gitApiObject = await getAzureDevOpsProvider(orgUrl, token)
+  const getCommits = async projectName => {
+    const repos = await gitApiObject.getRepositories(projectName)
+    const repoNames = repos.map(repo => repo.name)
+    return (
+      await Promise.all(
+        repoNames.map(repoName =>
+          gitApiObject.getCommits(
+            repoName,
+            {
+              fromDate: '01/01/2022 12:00:00 AM'
+            },
+            projectName,
+            0,
+            150000
+          )
+        )
+      )
+    ).flat()
+  }
+  const commits = (
+    await Promise.all(projects.map(project => getCommits(project)))
+  ).flat()
+  if (commits && commits.length) {
+    let commiters = {}
+    for (const commit of commits) {
+      if (commit) {
+        const author = commit?.author.name
```

Comment:
do we want the [author](https://github.com/microsoft/azure-devops-node-api/blob/e3daeea0d23b9d011a88b120e53eb4f8aa475542/api/interfaces/GitInterfaces.ts#L815) or [committer](https://github.com/microsoft/azure-devops-node-api/blob/e3daeea0d23b9d011a88b120e53eb4f8aa475542/api/interfaces/GitInterfaces.ts#L839) of the commit?

Reply:
actually I think we do want the author:

> The author is the person who originally wrote the code. The committer, on the other hand, is assumed to be the person who committed the code on behalf of the original author. This is important in Git because Git allows you to rewrite history, or apply patches on behalf of another person.